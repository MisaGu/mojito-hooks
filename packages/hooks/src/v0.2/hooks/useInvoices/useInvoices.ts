import { EMojitoKey } from '../../domain/enums/state.enum';
import { MojitoGetMyInvoicesResponse } from '../../domain/interfaces';
import { BaseQueryHookProps } from '../../domain/interfaces/hooks.interface';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';

function transformFn(response?: MojitoGetMyInvoicesResponse) {
  if (!response) return undefined;

  return response.getMyInvoices;
}

export type UseInvoicesData = ReturnType<typeof transformFn>;

export type UseInvoicesReturn = ReturnType<typeof useInvoices>;

export type UseInvoicesProps = BaseQueryHookProps<UseInvoicesData>;

export function useInvoices({ options }: UseInvoicesProps = {}) {
  return useMojitoFactory({
    as: 'invoices',
    query: EMojitoKey.invoices,
    options,
    transformFn,
    onlyAuthenticated: true,
  });
}

export default useInvoices;
