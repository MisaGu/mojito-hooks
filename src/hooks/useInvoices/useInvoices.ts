import { QueryKey } from '../../domain/utils/queryKeyFactory.util';
import { EMojitoKey } from '../../domain/enums/state.enum';
import { MojitoGetMyInvoicesResponse } from '../../domain/interfaces';
import { BaseQueryHookProps } from '../../domain/interfaces/hooks.interface';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';

function selectorFn(response?: MojitoGetMyInvoicesResponse) {
  if (!response) return undefined;

  return response.getMyInvoices;
}

export type UseInvoicesData = ReturnType<typeof selectorFn>;

export type UseInvoicesReturn = ReturnType<typeof useInvoices>;

export type UseInvoicesProps = BaseQueryHookProps<UseInvoicesData>;

export function useInvoices({ options }: UseInvoicesProps = {}) {
  return useMojitoFactory({
    as: 'invoices',
    queryKey: QueryKey.get(EMojitoKey.invoices),
    options,
    selectorFn,
    onlyAuthenticated: true,
  });
}

export default useInvoices;
