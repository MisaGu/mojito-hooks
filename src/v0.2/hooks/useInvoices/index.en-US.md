---
nav:
  path: /hooks
  group: Mojito
---

# useInvoices

Returns all invoices for the current user, including data for all the items in them.

## Examples

<code src="./demo/demo1.tsx" />
