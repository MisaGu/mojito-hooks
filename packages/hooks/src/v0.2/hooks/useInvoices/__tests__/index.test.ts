import { useMyInvoices } from '../useInvoices';

describe('useToggle', () => {
  it('should be defined', () => {
    expect(useMyInvoices).toBeDefined();
  });
});
