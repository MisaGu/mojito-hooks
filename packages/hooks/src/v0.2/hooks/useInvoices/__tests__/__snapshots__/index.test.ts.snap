// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useInvoices() returns the right result when needed 1`] = `
Object {
  "error": null,
  "invoices": Array [
    Object {
      "invoiceCreatedAt": "2022-05-23T00:18:12Z",
      "invoiceID": "mock-invoice-id-1",
      "invoiceNumber": 7030,
      "items": Array [
        Object {
          "buyersPremium": 0,
          "collectionItemId": "mock-collection-item-id",
          "collectionItemTitle": "Mock Collection Item",
          "collectionTitle": "",
          "destinationAddress": "0x1646BB353Ce7E259fB1BaC65Da64b469b648b5Da",
          "overheadPremium": 0,
          "saleDate": "2022-05-23T00:18:12Z",
          "salesTaxRate": 0,
          "taxes": 0,
          "totalPrice": 100,
          "unitPrice": 100,
          "units": 1,
        },
      ],
      "status": "Expired",
    },
  ],
  "isLoading": false,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": true,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
